src/inmanta/deploy/scheduler.py

```diff
@@ -670,6 +673,8 @@ async def report_resource_state(
                     state.deployment_result = deployment_result
                 return
 
+            if skipped_for_dependency:
+                state.blocked = BlockedStatus.TRANSIENT
```

Comment:
Finally, we should also drop it from `self._state.dirty` (already done below when the resource deploys successfully but should now also be done if we block the resource).

The drawback is that we also have to correctly unset the blocked status (back to `NO` and mark it as dirty again) when appropriate, because otherwise we'll never deploy it anymore. This part is missing from your PR atm.

I can think of two cases to consider:
1. new version comes in with one or more requires dropped
2. a dependency deploys successfully. We may choose to delegate this one to @Hugo-Inmanta's PR because I believe he already calculates this set. If we do, we should also delegate the "mark as dirty" so we don't get a stuck orchestrator in between both PRs. I'll leave it up to you to determine who does which part, considering double work and merge conflicts.

Reply:
I tried to follow this logic. I don't know if this what you have in mind. I'll add some comments

Reply:
We can also have a call about this if you prefer.

Reply:
Regarding this: 
>a dependency deploys successfully. We may choose to delegate this one to @Hugo-Inmanta's PR because I believe he already calculates this set. If we do, we should also delegate the "mark as dirty" so we don't get a stuck orchestrator in between both PRs. I'll leave it up to you to determine who does which part, considering double work and merge conflicts.

we should also delegate the part where we drop from the dirty set, right?

Reply:
Indeed. I see that I wasn't very clear on that.